Add build documentation (#55)
rapidsai/build-planning: docs/docs/release.md

# Release

This section covers miscellaneous tasks to do with the release.

> ## ptxcompiler and cubinlinker

Reviewer: Maybe a higher level section here for JIT tools?

```
## JIT tools

JIT tools help binaries be more portable and extend to new GPUs that didn't exist at the time the code was compiled. https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#binary-compatibility

### nvjitlinker (CUDA 12+)

...

### ptxcompiler and cubinlinker (CUDA 11)

...
```

In addition to the standard RAPIDS packages, RAPIDS also published `ptxcompiler` and `cubinlinker` Python packages.

> These are CUDA 11-specific tools to enable [MVC](https://docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html#cuda-compatibility-developer-s-guide) for JIT-compiled code using numba.

Reviewer: "MVC" is not on this linked page. Can you define the acronym? Maybe it has a new name, or has changed location?

[ptxcompiler is an open-source but deprecated package](https://github.com/rapidsai/ptxcompiler), scheduled for removal when RAPIDS drops CUDA 11 support.
`cubinlinker`, on the other hand, is a private repository maintained internally due to its use of CUDA internals.

At present, conda packages for `ptxcompiler` are built and published via [this conda-forge feedstock](https://github.com/conda-forge/ptxcompiler-feedstock).
conda packages for `cubinlinker` have generally been built manually and uploaded.
Wheels for both `ptxcompiler` and `cubinlinker` are built and uploaded using [this script](https://github.com/rapidsai/pypi-wheel-scripts/blob/main/cibuildwheel-local-scripts/cibuildwheel_ptxcompiler_cubinlinker.sh) in the `pypi-wheel-scripts` repository.

## pynvjitlink

> The CUDA 12 replacement for `ptxcompiler` and `cubinlinker` is [pynvjitlink](https://github.com/rapidsai/pynvjitlink/), which uses the public CUDA [nvjitlink](https://docs.nvidia.com/cuda/nvjitlink/index.html) APIs for the purpose of enabling MVC with PTX code.

Reviewer: ```suggestion
[pynvjitlink](https://github.com/rapidsai/pynvjitlink/) is a CUDA 12+ JIT tool that uses the public CUDA [nvjitlink](https://docs.nvidia.com/cuda/nvjitlink/index.html) APIs for the purpose of enabling [MVC](https://docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html#cuda-compatibility-developer-s-guide) with PTX code.
```
